postgresql/resource_postgresql_role.go

```diff
@@ -173,6 +184,30 @@ func resourcePostgreSQLRole() *schema.Resource {
 				Optional:    true,
 				Description: "Role to switch to at login",
 			},
+			roleParameterAttr: {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "Configuration parameters",
+				Elem:        resourcePostgreSQLRoleConfigurationParameter(),
+			},
+		},
+	}
+}
+
+func resourcePostgreSQLRoleConfigurationParameter() *schema.Resource {
```

Comment:
This is defined as a separate factory function instead of inlined in `resourcePostgreSQLRole()` because I need to get my hands on the resource when initializing a `Set` of them in `readRoleParameters`.